packages/textfield/src/mwc-textfield-base.ts

```diff
@@ -16,25 +16,42 @@ limitations under the License.
 */
 import '@material/mwc-notched-outline';
 
+import {MDCLineRippleFoundation} from '@material/line-ripple/foundation.js';
 import {addHasRemoveClass, classMap, FormElement, html, property, PropertyValues, query, TemplateResult} from '@material/mwc-base/form-element.js';
 import {floatingLabel, FloatingLabel} from '@material/mwc-floating-label';
 import {lineRipple, LineRipple} from '@material/mwc-line-ripple';
 import {NotchedOutline} from '@material/mwc-notched-outline';
 import {MDCTextFieldAdapter, MDCTextFieldInputAdapter, MDCTextFieldLabelAdapter, MDCTextFieldLineRippleAdapter, MDCTextFieldOutlineAdapter, MDCTextFieldRootAdapter} from '@material/textfield/adapter.js';
+import {MDCTextFieldCharacterCounterFoundation} from '@material/textfield/character-counter/foundation.js';
 import MDCTextFieldFoundation from '@material/textfield/foundation.js';
 import {ifDefined} from 'lit-html/directives/if-defined.js';
 
 import {characterCounter, CharacterCounter} from './character-counter/mwc-character-counter-directive.js';
 
+// must be done to get past lit-analyzer checks
+declare global {
+  interface Element {
+    foundation?: MDCLineRippleFoundation|MDCTextFieldCharacterCounterFoundation;
```

Comment:
Can we use more specific names so that we don't get collisions in the types? E.g. the next time another package has to augment element with `foundation`, they won't be able to.